packages/middleware/src/apiClientFactory/index.ts

```diff
@@ -10,9 +11,22 @@ import {
   CreateApiClientFn,
   ExtensionHookWith,
   ExtensionWith,
+  FnMarkedWithExtensionName,
 } from "../types";
 import { applyContextToApi } from "./applyContextToApi";
 
+/**
+ * Function marking endpoint added or overwritten by extension's extendApiMethod hook
+ * with information about source extension's name
+ */
+function markWithExtensionName(
+  apiMethod: FnMarkedWithExtensionName,
```

Comment:
Isnt FnMarkedWithExtensionName too complex? it is all about adding prop to object (fn is an object). Should we care what do you pass?

Reply:
It's useful. It prevents us from passing primitive values and unlocks only specified fields. We could rename it `CustomEndpointHandler`.

I also considered `object` type as it's exactly for locking primitive values but it blocks `_extensionName` field.